notebooks/logistic-regression.ipynb

```diff
@@ -647,6 +647,108 @@
     "else:\n",
     "    print(f\"logLoss: {evaluator.evaluate(cpu_transformed_df)}\")"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Sparse Vectors"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def sparse_vectors_compat(EstimatorClass):\n",
+    "    from pyspark.ml.evaluation import MulticlassClassificationEvaluator\n",
+    "    from pyspark.ml.feature import CountVectorizer, RegexTokenizer\n",
+    "    from pyspark.sql import Row\n",
+    "    from sklearn.datasets import fetch_20newsgroups\n",
+    "\n",
+    "    # prepare dataframe\n",
+    "    try:\n",
+    "        twenty_train = fetch_20newsgroups(subset=\"train\", shuffle=True, random_state=42)\n",
+    "    except:\n",
+    "        print(\"Error fetching 20 newsgroup dataset\")\n",
+    "    X = twenty_train.data\n",
+    "    y = twenty_train.target.tolist()\n",
+    "\n",
+    "    data = [\n",
+    "        Row(\n",
+    "            label=y[i],\n",
+    "            weight=1.0,\n",
+    "            text=X[i],\n",
+    "        )\n",
+    "        for i in range(len(X))\n",
+    "    ]\n",
+    "    \n",
+    "    df = spark.createDataFrame(data)\n",
+    "    \n",
+    "    # convert text to sparse vector\n",
+    "    tokenizer = RegexTokenizer(inputCol=\"text\", outputCol=\"tokens\")\n",
+    "    df = tokenizer.transform(df)\n",
+    "    cv = CountVectorizer(inputCol=\"tokens\", outputCol=\"features\")\n",
+    "    cv_model = cv.fit(df)\n",
+    "    df = cv_model.transform(df)\n",
+    "\n",
+    "    df_train, df_test = df.randomSplit([0.8, 0.2])\n",
```

Comment:
Better to have same data/test/train split for gpu and cpu.   This will be different for each call.

Reply:
Added seed=0.